src/repositories/orders.repository.ts

```diff
@@ -0,0 +1,81 @@
+import { OrderEntity } from "../schemas/order.entity.js";
+import { randomUUID } from "crypto";
+import { CartEntity } from "../schemas/cart.entity.js";
+import { countTotal } from "../utils/utils.js";
+import { readFile, writeFile } from 'fs/promises';
+
+const ORDERS_FILE_PATH = 'src/db/orders.json';
+
+async function loadOrders(): Promise<OrderEntity[]> {
+  try {
+    const data = await readFile(ORDERS_FILE_PATH, 'utf-8');
+    return JSON.parse(data);
+  } catch (err) {
+    console.error('Error loading orders:', err);
+    return [];
+  }
+};
+
+async function saveOrders(orders: OrderEntity[]): Promise<void> {
+  try {
+    await writeFile(ORDERS_FILE_PATH, JSON.stringify(orders), "utf-8")
+    console.log(`Orders saved successfully!`)
+  } catch (error) {
+    console.error('Error saving orders:', error);
+  }
+};
+
+export const OrderRepository = {
+  getAll: async (): Promise<OrderEntity[]> => {
+    const orders = await loadOrders()
+    return orders;
+  },
+  getOne: async (orderId: string): Promise<OrderEntity> => {
+    const orders = await loadOrders();
+    const order = orders.find((p: OrderEntity) => p.id === orderId);
+    if (!order) {
+      const customError = {
+        status: 404,
+        message: `Order with this id: ${ orderId } not found`,
+      };
```

Comment:
For it to be your custom error, you should create new instance of you error class and throw it.